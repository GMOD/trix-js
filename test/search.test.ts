import Trix from '../src/index';
import { LocalFile, RemoteFile } from 'generic-filehandle';
// import fetch from 'node-fetch'

const ixxFile1: string = './test/testData/test1/myTrix.ixx';
const localIxx1 = new LocalFile(ixxFile1);
const ixFile1: string = './test/testData/test1/myTrix.ix';
const localIx1 = new LocalFile(ixFile1);
const trix1 = new Trix(localIxx1, localIx1);

const ixxFile2: string = './test/testData/test2/out.ixx';
const localIxx2 = new LocalFile(ixxFile2);
const ixFile2: string = './test/testData/test2/out.ix';
const localIx2 = new LocalFile(ixFile2);
const trix2 = new Trix(localIxx2, localIx2);

const ixxFile3: string = './test/testData/test3/out.ixx';
const localIxx3 = new LocalFile(ixxFile3);
const ixFile3: string = './test/testData/test3/out.ix';
const localIx3 = new LocalFile(ixFile3);
const trix3 = new Trix(localIxx3, localIx3);

const ixxFile4: string = './test/testData/test4/out.ixx';
const localIxx4 = new LocalFile(ixxFile4);
const ixFile4: string = './test/testData/test4/out.ix';
const localIx4 = new LocalFile(ixFile4);
const trix4 = new Trix(localIxx4, localIx4);

// describe(`Test the parsing of test1 ixx file`, () => {
//   it('Parse test1/myTrix.ixx', async () => {
//     const trix = new Trix(localIxx1, localIx1);
//     const ixx = await trix.index;
//     expect(ixx).toMatchSnapshot();
//   });
// });

// describe(`Test the parsing of test2 ixx file`, () => {
//   it('Parse test2/out.ixx', async () => {
//     const trix = new Trix(localIxx2, localIx2);
//     const ixx = await trix.index;
//     expect(ixx).toMatchSnapshot();
//   });
// });


// -----------------------------------------------------------------
// Search trix1:

describe('Test a search of test1 ix file', () => {
  const searchTerm: string = "this";
  it(`Search for \"${searchTerm}\" in test1/myTrix.ix`, async () => {
    const hitList = await trix1.search(searchTerm);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test1 ix file', () => {
  const searchTerm: string = "is";
  it(`Search for \"${searchTerm}\" in test1/myTrix.ix`, async () => {
    const hitList = await trix1.search(searchTerm);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test1 ix file', () => {
  const searchTerm: string = "id1";
  it(`Search for \"${searchTerm}\" in test1/myTrix.ix`, async () => {
    const hitList = await trix1.search(searchTerm);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test1 ix file', () => {
  const searchTerm: string = "nope";
  it(`Search for \"${searchTerm}\" in test1/myTrix.ix`, async () => {
    const hitList = await trix1.search(searchTerm);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test1 ix file', () => {
  const searchTerm: string = "fOR";
  it(`Search for \"${searchTerm}\" in test1/myTrix.ix`, async () => {
    const hitList = await trix1.search(searchTerm);
    expect(hitList).toMatchSnapshot();
  });
});

// ----------------------------------------------------------------------
// Search trix2:

describe('Test a search of test2 ix file', () => {
  const searchTerm: string = "FocAd";
  it(`Search for \"${searchTerm}\" in test2/out.ix`, async () => {
    const hitList = await trix2.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test2 ix file', () => {
  const searchTerm: string = "LINC";
  it(`Search for \"${searchTerm}\" in test2/out.ix`, async () => {
    const hitList = await trix2.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test2 ix file', () => {
  const searchTerm: string = "none";
  it(`Search for \"${searchTerm}\" in test2/out.ix`, async () => {
    const hitList = await trix2.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test2 ix file', () => {
  const searchTerm: string = "prickle";
  it(`Search for \"${searchTerm}\" in test2/out.ix`, async () => {
    const hitList = await trix2.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test2 ix file', () => {
  const searchTerm: string = "tim";
  it(`Search for \"${searchTerm}\" in test2/out.ix`, async () => {
    const hitList = await trix2.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test2 ix file', () => {
  const searchTerm: string = "znf8";
  it(`Search for \"${searchTerm}\" in test2/out.ix`, async () => {
    const hitList = await trix2.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});

describe('Test a search of test2 ix file', () => {
  const searchTerm: string = "enst";
  it(`Search for \"${searchTerm}\" in test2/out.ix`, async () => {
    const hitList = await trix2.search(searchTerm);
    expect(hitList).toMatchSnapshot();
  });
});



// --------------------------------------------------
// Search trix3:


describe('Test a search of test3 ix file', () => {
  const searchTerm: string = "focad";
  it(`Search for \"${searchTerm}\" in test3/out.ix`, async () => {
    const hitList = await trix3.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});


describe('Test a search of test3 ix file', () => {
  const searchTerm: string = "nothing";
  it(`Search for \"${searchTerm}\" in test3/out.ix`, async () => {
    const hitList = await trix3.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});


describe('Test a search of test3 ix file', () => {
  const searchTerm: string = "tim";
  it(`Search for \"${searchTerm}\" in test3/out.ix`, async () => {
    const hitList = await trix3.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});


describe('Test a search of test3 ix file', () => {
  const searchTerm: string = "prickle";
  it(`Search for \"${searchTerm}\" in test3/out.ix`, async () => {
    const hitList = await trix3.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});


describe('Test a search of test3 ix file', () => {
  const searchTerm: string = "oca";
  it(`Search for \"${searchTerm}\" in test3/out.ix`, async () => {
    const hitList = await trix3.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});


describe('Test a search of test3 ix file', () => {
  const searchTerm: string = "zzz";
  it(`Search for \"${searchTerm}\" in test3/out.ix`, async () => {
    const hitList = await trix3.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});

// ---------------------
describe('Test a search of test4 ix file', () => {
  const searchTerm: string = "ek";
  it(`Search for \"${searchTerm}\" in test4/out.ix`, async () => {
    const hitList = await trix4.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});




// ------------------------
// Test maxResults:

const trix3B = new Trix(localIxx3, localIx3, 5);

describe('Test maxResults for search of test3 ix file', () => {
  const searchTerm: string = "tim";
  it(`Search for \"${searchTerm}\" with a max of 5 results`, async () => {
    const hitList = await trix3B.search(searchTerm);
    // console.log(hitList);
    expect(hitList).toMatchSnapshot();
  });
});


