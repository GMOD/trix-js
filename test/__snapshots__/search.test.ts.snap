// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test a search of test1 ix file Search for "fOR" in test1/myTrix.ix 1`] = `
Array [
  "id1",
  "id2",
  "id3",
]
`;

exports[`Test a search of test1 ix file Search for "for this" in test1/myTrix.ix 1`] = `
Array [
  "id1",
  "id2",
  "id3",
]
`;

exports[`Test a search of test1 ix file Search for "id1" in test1/myTrix.ix 1`] = `
Array [
  "id1",
]
`;

exports[`Test a search of test1 ix file Search for "is" in test1/myTrix.ix 1`] = `
Array [
  "id1",
  "id2",
  "id3",
]
`;

exports[`Test a search of test1 ix file Search for "nope" in test1/myTrix.ix 1`] = `Array []`;

exports[`Test a search of test1 ix file Search for "this is for id2" in test1/myTrix.ix 1`] = `
Array [
  "id2",
]
`;

exports[`Test a search of test1 ix file Search for "this" in test1/myTrix.ix 1`] = `
Array [
  "id1",
  "id2",
  "id3",
]
`;

exports[`Test a search of test2 ix file Search for "FocAd" in test2/out.ix 1`] = `
Array [
  "ENST00000439876.1",
  "ENST00000445051.1",
]
`;

exports[`Test a search of test2 ix file Search for "LINC" in test2/out.ix 1`] = `
Array [
  "ENST00000416999.1",
  "ENST00000423414.5",
  "ENST00000429901.2",
  "ENST00000431189.1",
  "ENST00000433079.5",
  "ENST00000435523.5",
  "ENST00000443623.5",
  "ENST00000451786.5",
  "ENST00000642156.1",
  "ENST00000642483.1",
  "ENST00000642602.1",
  "ENST00000643135.1",
  "ENST00000643373.1",
  "ENST00000643855.1",
  "ENST00000643874.1",
  "ENST00000644188.1",
  "ENST00000644804.1",
  "ENST00000645248.1",
  "ENST00000646429.1",
  "ENST00000646587.1",
]
`;

exports[`Test a search of test2 ix file Search for "enst ac" in test2/out.ix 1`] = `
Array [
  "enst00000562538.1",
  "enst00000634116.1",
  "enst00000449652.1",
  "enst00000661649.1",
  "enst00000415249.1",
  "enst00000456270.1",
  "enst00000653602.1",
  "enst00000668236.1",
  "enst00000661747.1",
  "enst00000452326.1",
  "enst00000431090.1",
  "enst00000608816.1",
  "enst00000607934.1",
  "enst00000658236.1",
  "enst00000661478.1",
  "enst00000664918.1",
  "enst00000413812.1",
  "enst00000450061.5",
  "enst00000667616.1",
  "enst00000656084.1",
]
`;

exports[`Test a search of test2 ix file Search for "enst alg" in test2/out.ix 1`] = `
Array [
  "enst00000430794.1",
  "enst00000508969.2",
  "enst00000511954.5",
  "enst00000524714.1",
  "enst00000525473.1",
  "enst00000657407.1",
  "enst00000661578.1",
  "enst00000664227.1",
  "enst00000665956.1",
  "enst00000670269.1",
  "enst00000531305.1",
]
`;

exports[`Test a search of test2 ix file Search for "enst" in test2/out.ix 1`] = `
Array [
  "ENST00000229465.10",
  "ENST00000230113.5",
  "ENST00000235290.7",
  "ENST00000242109.5",
  "ENST00000244820.2",
  "ENST00000244906.6",
  "ENST00000248980.9",
  "ENST00000250776.5",
  "ENST00000250805.5",
  "ENST00000253838.3",
  "ENST00000253848.3",
  "ENST00000255183.8",
  "ENST00000262354.5",
  "ENST00000276770.8",
  "ENST00000276779.8",
  "ENST00000279067.3",
  "ENST00000289890.7",
  "ENST00000291374.11",
  "ENST00000292748.7",
  "ENST00000294715.6",
]
`;

exports[`Test a search of test2 ix file Search for "none" in test2/out.ix 1`] = `Array []`;

exports[`Test a search of test2 ix file Search for "prickle" in test2/out.ix 1`] = `
Array [
  "ENST00000460946.1",
  "ENST00000476308.1",
  "ENST00000484703.1",
  "ENST00000473434.1",
  "ENST00000487097.1",
  "ENST00000659263.1",
]
`;

exports[`Test a search of test2 ix file Search for "tim" in test2/out.ix 1`] = `
Array [
  "ENST00000429104.1",
  "ENST00000444438.2",
  "ENST00000651208.1",
  "ENST00000651763.1",
]
`;

exports[`Test a search of test2 ix file Search for "znf8" in test2/out.ix 1`] = `
Array [
  "ENST00000591325.1",
  "ENST00000443154.3",
  "ENST00000592296.5",
]
`;

exports[`Test a search of test3 ix file Search for "focad" in test3/out.ix 1`] = `
Array [
  "ENSMUST00000097992.10",
  "ENSMUST00000107147.2",
  "ENSMUST00000107148.8",
  "ENSMUST00000129967.2",
  "ENSMUST00000130603.2",
  "ENSMUST00000144388.2",
  "ENSMUST00000147527.2",
  "ENSMUST00000159342.8",
  "ENSMUST00000161058.8",
  "ENSMUST00000162039.2",
]
`;

exports[`Test a search of test3 ix file Search for "nothing" in test3/out.ix 1`] = `Array []`;

exports[`Test a search of test3 ix file Search for "oca" in test3/out.ix 1`] = `
Array [
  "ENSMUST00000032633.12",
  "ENSMUST00000144739.8",
  "ENSMUST00000152693.8",
  "ENSMUST00000154559.2",
  "ENSMUST00000155533.2",
  "ENSMUST00000156886.8",
]
`;

exports[`Test a search of test3 ix file Search for "prickle" in test3/out.ix 1`] = `
Array [
  "ENSMUST00000048982.11",
  "ENSMUST00000109255.3",
  "ENSMUST00000032093.12",
  "ENSMUST00000113445.2",
  "ENSMUST00000113446.8",
  "ENSMUST00000113447.8",
  "ENSMUST00000033485.14",
  "ENSMUST00000123172.3",
  "ENSMUST00000129662.3",
  "ENSMUST00000141876.2",
  "ENSMUST00000153994.2",
  "ENSMUST00000113299.6",
  "ENSMUST00000113300.8",
  "ENSMUST00000142014.2",
  "ENSMUST00000151540.2",
  "ENSMUST00000152214.4",
]
`;

exports[`Test a search of test3 ix file Search for "tim" in test3/out.ix 1`] = `
Array [
  "ENSMUST00000055102.13",
  "ENSMUST00000109225.9",
  "ENSMUST00000125008.2",
  "ENSMUST00000169584.8",
  "ENSMUST00000068877.7",
  "ENSMUST00000055539.11",
  "ENSMUST00000105240.8",
  "ENSMUST00000105242.8",
  "ENSMUST00000105243.9",
  "ENSMUST00000105244.8",
  "ENSMUST00000105245.3",
  "ENSMUST00000125289.8",
  "ENSMUST00000129147.2",
  "ENSMUST00000135376.2",
  "ENSMUST00000135538.2",
  "ENSMUST00000136854.2",
  "ENSMUST00000142149.2",
  "ENSMUST00000142484.2",
  "ENSMUST00000145710.8",
  "ENSMUST00000146945.2",
]
`;

exports[`Test a search of test3 ix file Search for "zzz" in test3/out.ix 1`] = `
Array [
  "ENSMUST00000089982.11",
  "ENSMUST00000106100.9",
  "ENSMUST00000106101.2",
  "ENSMUST00000106103.8",
  "ENSMUST00000135044.3",
  "ENSMUST00000138115.8",
  "ENSMUST00000150802.8",
  "ENSMUST00000197987.5",
  "ENSMUST00000198468.2",
  "ENSMUST00000200570.5",
]
`;

exports[`Test a search of test4 ix file Search for "ek" in test4/out.ix 1`] = `
Array [
  "id11",
  "id8",
  "id12",
  "id13",
  "id14",
  "id46",
  "id39",
  "id41",
  "id16",
  "id1",
  "id17",
  "id9",
  "id18",
  "id19",
  "id10",
  "id2",
  "id20",
  "id26",
  "id4",
  "id44",
]
`;

exports[`Test maxResults for search of test3 ix file Search for "tim" with a max of 5 results 1`] = `
Array [
  "ENSMUST00000055102.13",
  "ENSMUST00000109225.9",
  "ENSMUST00000125008.2",
  "ENSMUST00000169584.8",
  "ENSMUST00000068877.7",
]
`;
